File: signature_demo/nodejs/signature.js
const crypto = require('crypto');

const timestamp = '1562952827927000'; // please use the same timestamp (in milliseconds) as in request header
const api_path = 'user/info';
const msg = [timestamp, '+', api_path].join('');
const apiSecret = '98dRnVqbIhriRMZJfax3EvSPRBJrIuZ6J755KcLz6napGUpzmApJmfBY7EY3yYZn';

function signature(query_string) {
    return crypto
        .createHmac('sha256', apiSecret)
        .update(query_string)
        .digest('base64');
}

console.log("prehash msg: ");
console.log(msg);
console.log("signature:");
console.log(signature(msg));

console.log("\n");

File: signature_demo/golang/signature.go
package main

import (
	"crypto/hmac"
	"crypto/sha256"
	"encoding/base64"
	"fmt"
	"strings"
	"strconv"
)


func sign(message string, secret string) string {
	key := []byte(secret)
	h := hmac.New(sha256.New, key)
	h.Write([]byte(message))
	return base64.StdEncoding.EncodeToString(h.Sum(nil))
}

func main() {

	secret := "98dRnVqbIhriRMZJfax3EvSPRBJrIuZ6J755KcLz6napGUpzmApJmfBY7EY3yYZn"
	timestamp := 1562952827927000
	ts := strconv.Itoa(timestamp)
	apiPath := "user/info"
	msg := strings.Join([]string{ts, "+", apiPath}, "")
	fmt.Println("Prehash msg: ", msg)
	signature := sign(msg, secret)
	fmt.Println("Signature: ", signature)
}

